src/main/java/uk/gov/justice/digital/service/DeltaLakeService.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.justice.digital.service;
+
+import io.delta.tables.DeltaTable;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+
+public class DeltaLakeService {
+
+    public boolean exists(final String prefix, final String schema, final String table) {
+        return DeltaTable.isDeltaTable(getTablePath(prefix, schema, table));
+    }
+
+    private String getTablePath(final String prefix, final String schema, final String table) {
+        return String.join("/", prefix, schema, table);
+    }
+
+    public void append(final String prefix, final String schema, final String table, final Dataset<Row> df) {
+        df.write()
+                .format("delta")
+                .mode("append")
+                .option("path", getTablePath(prefix, schema, table))
+                .save();
+    }
+
+    public void replace(final String prefix, final String schema, final String table, final Dataset<Row> df) {
+        df.write()
+                .format("delta")
+                .mode("overwrite")
+                .option("overwriteSchema", true)
+                .option("path", getTablePath(prefix, schema, table))
+                .save();
+    }
+
+    public void delete(final String prefix, final String schema, final String table) {
+        System.out.println("deleting table.");
+        final DeltaTable deltaTable = getTable(prefix, schema, table);
+        if(deltaTable != null) {
```

Comment:
Do we care if the table is null? If this is meaningful consider logging a warning.